File: LexerPLSharpVersion/LexerPLSharpVersion/constants/KeyWords.cs
using System;
using LexerPLSharpVersion.token;

namespace LexerPLSharpVersion.constants {
   public static class KeyWords {
      private const string Program = "program";
      private const string Procedure = "procedure";
      private const string Call = "call";
      private const string Begin = "begin";
      private const string End = "end";
      private const string If = "if";
      private const string Then = "then";
      private const string Else = "else";
      private const string Do = "do";
      private const string While = "while";
      private const string Var = "var";
      private const string Const = "const";

      public static TokenType DetectKeyWordType(string value) {
         switch (value.ToLower()){
            case Program:
               return TokenType.Program;
            case Procedure:
               return TokenType.Procedure;
            case Call:
               return TokenType.Call;
            case Begin:
               return TokenType.Begin;
            case End:
               return TokenType.End;
            case If:
               return TokenType.If;
            case Then:
               return TokenType.Then;
            case Else:
               return TokenType.Else;
            case While:
               return TokenType.While;
            case Do:
               return TokenType.Do;
            case Var:
               return TokenType.Var;
            case Const:
               return TokenType.Const;
            default:
               return TokenType.Identifier;
         }
      }
   }
}

File: OwnProgrammingLanguage/src/main/java/lexer/constants/KeyWords.java
package lexer.constants;

import token.TokenType;

public class KeyWords {
    private final static String PROGRAM = "program";
    private final static String PROCEDURE = "procedure";
    private final static String CALL = "call";
    private final static String BEGIN = "begin";
    private final static String END = "end";
    private final static String IF = "if";
    private final static String THEN = "then";
    private final static String ELSE = "else";
    private final static String WHILE = "while";
    private final static String DO = "do";
    private final static String VAR = "var";
    private final static String CONST = "const";
    private final static String ODD = "odd";

    public static TokenType detectKeyWordType(String value) {
        switch (value.toLowerCase()) {
            case PROGRAM:
                return TokenType.PROGRAM;
            case PROCEDURE:
                return TokenType.PROCEDURE;
            case CALL:
                return TokenType.CALL;
            case BEGIN:
                return TokenType.BEGIN;
            case END:
                return TokenType.END;
            case IF:
                return TokenType.IF;
            case THEN:
                return TokenType.THEN;
            case ELSE:
                return TokenType.ELSE;
            case WHILE:
                return TokenType.WHILE;
            case DO:
                return TokenType.DO;
            case VAR:
                return TokenType.VAR;
            case CONST:
                return TokenType.CONST;
            case ODD:
                return TokenType.ODD;
            default:
                return TokenType.IDENTIFIER;
        }
    }
}
